buildSrc/subprojects/jvm/src/main/kotlin/gradlebuild/propagated-env-variables.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gradlebuild
+
+import org.gradle.api.tasks.testing.Test
+import gradlebuild.basics.BuildEnvironment
+
+
+val propagatedEnvAllowList = listOf(
+    // Otherwise Windows complains "Unrecognized Windows Sockets error: 10106"
+    "SystemRoot",
+    "OS",
+
+    "ANDROID_HOME",
+
+    // Used by Visual Studio
+    "USERNAME",
+    "USER",
+    "USERDOMAIN",
+    "USERPROFILE",
+
+    // Used by Gradle test infrastructure
+    "REPO_MIRROR_URL",
+
+    // temp dir
+    "TMPDIR",
+    "TMP",
+    "TEMP",
+
+    // For Build Distribution build
+    "CI",
+    "GRADLE_ENTERPRISE_ACCESS_KEY"
```

Comment:
See this build: https://builds.gradle.org/viewLog.html?buildId=36905741&tab=buildResultsDiv&buildTypeId=Gradle_Check_BuildDistributions

Our build has a lot of `CI`-specific logic, so I think it would be better to keep "the inner build" consistent with the "outer build".